### PR TITLE
Increase precision of taxes and quantity

### LINKED ISSUE #96: FEATURE: Higher precision for percentage based inputs
17.5% = 0.175 or 20.25 = 0.2025 should be able to be processed

### LINKED ISSUE #97: Higher precision for InvoiceItem.quantity
It should be possible to add a decimal number for a quantity.

Think of hourly prices or journey price which could have a quantity of 4.2h or 133.7km.

### PR DESCRIPTION
**Pull request by iTitus**
Closes #96
Closes #97

### CODE REVIEW

File: invoice/models.py [hunk 1]
```diff
@@ -153,14 +153,15 @@ class InvoiceItem(Model):
     """Line item of an invoice."""
     name = CharField(max_length=120)
     description = CharField(max_length=1000)
-    quantity = IntegerField(
-        validators=[MinValueValidator(0), MaxValueValidator(MAX_VALUE_DJANGO_SAVE)])
+    quantity = DecimalField(max_digits=19, decimal_places=2,
```
Comment (Segelzwerg) [inline invoice/models.py:156]: I think we can allow for more decimal places. I'm thinking of 0.001 kg.
So probably something like 5 should do for most use case.

File: invoice/models.py [hunk 2]
```diff
@@ -176,10 +177,21 @@ def total(self) -> Decimal:
...
+                self.quantity_string,
+                f'{self.price:.2f}',
+                self.tax_string,
+                f'{self.net_total:.2f}',
+                f'{self.total:.2f}']
+
+    @property
+    def quantity_string(self) -> str:
+        """Get the quantity string."""
+        return f'{self.quantity:.2f}'.rstrip('0').rstrip('.,')
```
Comment (Segelzwerg) [inline invoice/models.py:191]: See above.

Comment (Segelzwerg) [inline invoice/models.py:191]: Why do you allow above 4 decimals places but here only 2?

**Review by Segelzwerg — CHANGES_REQUESTED** [review]

File: invoice/models.py (hunk 2 above)
Comment (iTitus) [reply invoice/models.py:191]: quantity had 2 decimal places, so we cut to two here.
taxes has 4, but is multiplied by 100, so two should cover all significant digits

**Pull request merged**